distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -2447,6 +2453,82 @@ private void checkExpectations(CometDClient cometDClient, Integer numNotificatio
 
     }
 
+    /**
+     * Confirms that the "download resource to local cache" action's url can be
+     * found in the CometDClient messages, and compared to the expected url.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void findMetacardActionUrlUsingCometDClient() throws Exception {
+
+       String metacardId = ingestXmlWithProduct(testName.getMethodName() + ".txt");
+
+        metacardsToDelete.add(metacardId);
+
+        final String RESPONSE_CHANNEL = "/0193d9e7f9ed4f8f8bd02103143c41d6";
+
+       cometDClient = setupCometDClient(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL,
+                QUERY_PUBLISH_CHANNEL,
+                RESPONSE_CHANNEL));
+
+        String src = "ddf.distribution";
+
+        makeAndPublishQueryData(RESPONSE_CHANNEL, src, cometDClient, metacardId);
+
+        expect("Waiting for response").within(20, SECONDS)
+                .until(() -> cometDClient.getMessages(RESPONSE_CHANNEL)
+                        .size() >= 1);
+
+        List<String> messages = cometDClient.getAllMessages();
+
+        LOGGER.debug("cometDClient messages size: " + messages.size());
```

Comment:
Same here on slf4j logging. Basically you can do `cometDClient messages size: {}, messages.size()`